yt_dlp/extractor/bilibili.py

```diff
@@ -535,6 +539,83 @@ def _real_extract(self, url):
         return self.playlist_result(self._entries(list_id), list_id)
 
 
+# https://www.bilibili.com/v/kichiku/mad/#/
+class BilibiliCategoryIE(SearchInfoExtractor):
+    IE_NAME = "Bilibili category extractor"
+    _MAX_RESULTS = 1000000
+    _VALID_URL = r'https?:\/\/www\.bilibili\.com\/v\/[a-zA-Z]+\/[a-zA-Z]+'
+    _TESTS = [{
+        'url': 'https://www.bilibili.com/v/kichiku/mad',
+        'info_dict': {
+            'id': 'kichiku: mad',
+        },
+        'playlist_maxcount': 45,
+        'params': {
+            "playlist_end": '45'
+        }
+    }]
+
+    def _entries(self, category, subcategory, query):
+        # map of categories : subcategories : RIDs
+        rid_map = {
+            "kichiku": {
+                "mad": 26,
+                "manual_vocaloid": 126,
+                "guide": 22,
+                "theatre": 216,
+                "course": 127
+            },
+        }
+
+        if category not in rid_map:
+            raise ExtractorError("The supplied category, %s, is not supported" % category)
+
+        if subcategory not in rid_map[category]:
+            raise ExtractorError("The supplied subcategory, %s, is not supported" % subcategory)
+
+        rid_value = rid_map[category][subcategory]
+
+        api_url = "https://api.bilibili.com/x/web-interface/newlist?rid=%d&type=1&ps=20&jsonp=jsonp" % rid_value
+        json_str = self._download_webpage(api_url + "&pn=1", "None", query={"Search_key": query})
+        parsed_json = json.loads(json_str)
+        page_data = try_get(parsed_json, lambda x: x['data']['page'], dict)
+        num_pages = math.ceil(page_data['count'] / page_data['size'])
+
+        num_extracted = 0
+
+        # desc order
+        for page_number in range(1, num_pages + 1):
+            time.sleep(2)
+
+            json_str =  self._download_json(
+                api_url + "&pn=%s" % page_number,'None',  query={"Search_key": query},
+                note='Extracting results from page %s of %s' % (page_number, num_pages))
+
+            # Ascending by publish date
+            video_list = sorted(parsed_json['data']['archives'], key=lambda video: video['pubdate'])
+            video_list_processed = list(map(lambda video: self.url_result("https://www.bilibili.com/video/%s" % video['bvid'],
+                                                        'BiliBili', video['bvid']), video_list))
+            for video in video_list_processed:
+                num_extracted += 1
+                yield video
+
+                if num_extracted >= BilibiliCategoryIE._MAX_RESULTS:
+                    return
+
+    @classmethod
+    def _make_valid_url(cls):
+        return cls._VALID_URL
+
+    def _real_extract(self, url):
+        u = compat_urllib_parse_urlparse(url)
+        category = u.path.split('/')[2]
+        subcategory = u.path.split('/')[3]
```

Comment:
```suggestion
        u = compat_urllib_parse_urlparse(url)
        category, subcategory = u.path.split('/')[2:3]
```
or should use regex groups